doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,724 @@
+commands that want to open in the same window _need_ to specify `new-tab`,
+otherwise they'll all appear in new windows.
+
+If it's `new-tab`, then how do `--initialRows` (etc) work? `new-tab` generally
+_doesn't_ accept those parameters, because it's going to be inheriting the
+parent's window size. Do we just ignore them for subsequent invocations? I
+suppose that makes sense, once the first tab has set those, then the other tabs
+can't really change them.
+
+When dealing with a file full of startup commands, we'll assume all of them are
+intended for the given window. So the first `new-tab` in the file will create
+the window, and all subsequent `new-tab` commands will create tabs in that same
+window.
+ -->
+
+### Options
+
+#### `--help,-h,-?,/?,`
+Runs the `help` command.
+
+#### `--version,-v`
+Runs the `version` command.
+
+#### `--session,-s session-id`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. This feature is dependent upon
+other planned work landing, so is only provided as an example, of what it might
+look like. See [Future Considerations](#Future-Considerations) for more details.
+
+#### `--file,-f configuration-file`
+Run these commands in the given Windows Terminal session. Enables opening new
+tabs in already running Windows Terminal windows. See [Future
+Considerations](#Future-Considerations) for more details.
+
+### Commands
+
+#### `help`
+
+`help`
+
+Display the help message
+
+#### `version`
+
+`version`
+
+Display version info for the Windows Terminal
+
+#### `open-settings`
+
+`open-settings [--defaults,-d]`
+
+Open the settings file. If this command is provided alone, it does not open the
+terminal window.
+
+**Parameters**:
+* `--defaults,-d`: Open the `defaults.json` file instead of the `profiles.json`
+  file.
+
+#### `list-profiles`
+
+`list-profiles [--all,-A] [--showGuids,-g]`
+
+Displays a list of each of the available profiles. Each profile displays it's
+name, seperated by newlines.
+
+**Parameters**:
+* `--all,-A`: Show all profiles, including profiles marked `"hidden": true`.
+* `--showGuids,-g`:  In addition to showing names, also list each profile's
+  guid. These GUIDs should probably be listed _first_ on each line, to make
+  parsing output easier.
+
+#### `new-tab`
+
+`new-tab [--initialPosition x,y]|[--maximized]|[--fullscreen] [--initialRows rows] [--initialCols cols] [terminal_parameters]`
+
+Opens a new tab with the given customizations. On it's _first_ invocation, also
+opens a new window. Subsequent `new-tab` commands will all open new tabs in the
+same window.
+
+**Parameters**:
+* `--initialPosition x,y`: Create the new Windows Terminal window at the given
+  location on the screen in pixels. This parameter is only used when initially
+  creating the window, and ignored for subsequent `new-tab` commands. When
+  combined with any of `--maximized` or `--fullscreen`, TODO: what do?

```

Comment:
Need to iron out the "what do"

If we do not support these arguments today, move initial position into future consideration

Reply:
but generally, these override and work in the same way the config file does: fullscreen/max win.

Reply:
actually fullscreen and max _use the monitor specified by initialPos_

Reply:
point to their spec if you want

Reply:
Lol I just updated this to say that CLI11 will reject combinations of these _for us_, which I think is maybe more correct. I can always update the spec when I do add those features, if we really want them.